core/src/main/java/org/apache/accumulo/fate/Fate.java

```diff
@@ -71,24 +83,51 @@ public void run() {
         try {
           tid = store.reserve();
           TStatus status = store.getStatus(tid);
+          if (status == FAILED) {
+            runnerLog.info(
+                "FATE txid " + Long.toHexString(tid) + " likely cancelled before it started.");
+            return;
+          }
           Repo<T> op = store.top(tid);
-          if (status == TStatus.FAILED_IN_PROGRESS) {
+          if (status == FAILED_IN_PROGRESS) {
             processFailed(tid, op);
           } else {
             Repo<T> prevOp = null;
+            boolean exceptionSeen = false;
             try {
-              deferTime = op.isReady(tid, environment);
-
-              if (deferTime == 0) {
-                prevOp = op;
-                if (status == TStatus.SUBMITTED) {
-                  store.setStatus(tid, TStatus.IN_PROGRESS);
+              try {
+                synchronized (RUNNING_TRANSACTIONS) {
+                  RUNNING_TRANSACTIONS.put(tid, Thread.currentThread());
```

Comment:
```suggestion
                  // If this ever fails its indication that there is a concurrency problem in other code because its not  expected that multiple threads would be working on the same transaction id at the same time.
                  Preconditions.checkState(RUNNING_TRANSACTIONS.putIfAbsent(tid,Thread.currentThread()) == null);
```

Reply:
This is OBE as I modified the code in 7b7d1b7 to only cancel new or submitted transactions.